packages/@icon-magic/distribute/src/create-custom-element.ts

```diff
@@ -0,0 +1,48 @@
+import { Asset, Icon } from '@icon-magic/icon-models';
+import * as fs from 'fs-extra';
+import * as path from 'path';
+
+import { kebabToCamel } from './utils';
+
+/**
+ * Saves custom element assets as js files
+ * @param assets custom element assets to be saved
+ * @param outputPath path to write to
+ * @param icon from which custom element assets are derived
+ * @param doNotRemoveSuffix boolean, when true will keep the "-mixed" and
+ *        "-with-image" suffix in file name
+ */
+export async function createCustomElement(
+  assets: Asset[],
+  outputPath: string,
+  icon: Icon,
+  doNotRemoveSuffix: boolean
+): Promise<void> {
+  for (const asset of assets) {
+    const prefix = icon?.generatedMetadata?.customElement?.namePrefix || '';
+    // file name must match the custom element name
+    const customElementName = `${prefix}${icon.iconName}-${asset.name}`;
+    let fileName = customElementName;
+    let fileContents = (await asset.getContents()) as string;
+
+    // Remove "-mixed" and "-with-image" suffix from the name. File will have same name as light version.
+    if (!doNotRemoveSuffix && asset.colorScheme === 'mixed') {
+      fileName = fileName.replace(/-mixed$/, '');
+      fileName = fileName.replace(/-with-image/, '');
+    }
+
+    // update custom element name in the file contents
+    fileContents = fileContents.toString();
+    let regex = new RegExp(customElementName, 'g');
+    fileContents = fileContents.replace(regex, fileName);
+    regex = new RegExp(kebabToCamel(customElementName), 'g');
+    fileContents = fileContents.replace(regex, kebabToCamel(fileName));
```

Comment:
Why not generate in camelCase to begin with?

Reply:
Kebabcase is for custom element name in the file contents, eg, `<icon-magic-home-filled>` while camelcase is for variable name in the example in JsDoc in the file contents, eg, `iconMagicHomeFilled()`.  [Element name cannot be camelcase](https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements#high-level_view)
> custom element names require a dash to be used in them (kebab-case); they can't be single words

Reply:
Why not do this at the generate step?

Reply:
Consistent with `hbs` distribute script, we want to give the option to the user to retain or strip out suffix `-mixed` and `-with-image` in the filename (custom element name inside the file) during the distribution phase. If we do this in generate phase, we'll lose the distinction whether the icon is mixed/with-image and moreover, mixed icon will override the light icon or vice-versa.